lib/gcloud/logging/logger.rb

```diff
@@ -0,0 +1,216 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+module Gcloud
+  module Logging
+    ##
+    # # Logger
+    #
+    # A (mostly) API compatible logger for ruby's Logger.
+    class Logger
+      ##
+      # @private The logging object.
+      attr_accessor :logging
+
+      ##
+      # @private The Google Cloud log_name to write the log entry with.
+      attr_reader :log_name
+
+      ##
+      # @private The Google Cloud resource to write the log entry with.
+      attr_reader :resource
+
+      ##
+      # @private The Google Cloud labels to write the log entry with.
+      attr_reader :labels
+
+      ##
+      # @private Creates a new Connection instance.
+      def initialize logging, log_name, resource, labels = nil
+        @logging = logging
+        @log_name = log_name
+        @resource = resource
+        @labels = labels
+        @level = 0 # DEBUG is the default behavior
+      end
+
+      ##
+      # Log a DEBUG message.
+      def debug message = nil, &block
+        if block_given?
+          add 0, nil, message, &block
+        else
+          add 0, message, nil, &block
+        end
+      end
+
+      ##
+      # Log an INFO message.
+      def info message = nil, &block
+        if block_given?
+          add 1, nil, message, &block
+        else
+          add 1, message, nil, &block
+        end
+      end
+
+      ##
+      # Log a WARN message.
+      def warn message = nil, &block
+        if block_given?
+          add 2, nil, message, &block
+        else
+          add 2, message, nil, &block
+        end
+      end
+
+      ##
+      # Log an ERROR message.
+      def error message = nil, &block
+        if block_given?
+          add 3, nil, message, &block
+        else
+          add 3, message, nil, &block
+        end
+      end
+
+      ##
+      # Log a FATAL message.
+      def fatal message = nil, &block
+        if block_given?
+          add 4, nil, message, &block
+        else
+          add 4, message, nil, &block
+        end
+      end
+
+      ##
+      # Log an UNKNOWN message. This will be printed no matter what the logger's
+      # level is.
+      def unknown message = nil, &block
+        if block_given?
+          add 5, nil, message, &block
+        else
+          add 5, message, nil, &block
+        end
+      end
+
+      ##
+      # Log a message if the given severity is high enough. This is the generic
+      # logging method. Users will be more inclined to use {#debug}, {#info},
+      # {#warn}, {#error}, and {#fatal}.
```

Comment:
Yes, but I think we need to revisit all the docs so I might leave that for when we take a more comprehensive pass on the documentation.